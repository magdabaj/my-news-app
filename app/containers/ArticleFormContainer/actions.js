/*
 *
 * ArticleFormContainer actions
 *
 */

import {ADD_ARTICLE, ADD_ARTICLE_CANCELED, ADD_ARTICLE_ERROR, ADD_ARTICLE_SUCCESS} from "./constants";

export const addArticle = (article, tagId, user) => ({
  type: ADD_ARTICLE,
  article,
  tagId,
  user
});

export const addArticleSuccess = article => ({
  type: ADD_ARTICLE_SUCCESS,
  article,
});

export const addArticleError = error => ({
  type: ADD_ARTICLE_ERROR,
  error,
});

export const addArticleCanceled = () => ({
  type: ADD_ARTICLE_CANCELED,
});