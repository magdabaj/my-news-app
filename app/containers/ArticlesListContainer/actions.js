/*
 *
 * ArticlesListContainer actions
 *
 */

import {
  GET_ALL_ARTICLES,
  GET_ARTICLES_FOR_TAG,
  GET_ARTICLES_FOR_TAG_FAILED,
  GET_ARTICLES_FOR_TAG_SUCCESS, START_ADD
} from "./constants";

export const getAllArticles = () => ({
  type: GET_ALL_ARTICLES
});

export const getArticlesForTag = tagId => ({
  type: GET_ARTICLES_FOR_TAG,
  tagId,
});

export const getArticlesForTagSuccess = articles => ({
  type: GET_ARTICLES_FOR_TAG_SUCCESS,
  articles,
});

export const getArticlesForTagFailed = error => ({
  type: GET_ARTICLES_FOR_TAG_FAILED,
  error,
});

export const startAdd = tagName => ({
  type: START_ADD,
  tagName,
});