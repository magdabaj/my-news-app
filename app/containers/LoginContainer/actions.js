/*
 *
 * LoginContainer actions
 *
 */

import {CANCEL_LOGIN, LOGIN, LOGIN_FAILED, LOGIN_SUCCESS} from "./constants";

export const login = (email, password) => ({
  type: LOGIN,
  email,
  password,
});

export const loginSuccess = user => ({
  type: LOGIN_SUCCESS,
  user,
});

export const loginFailed = error => ({
  type: LOGIN_FAILED,
  error,
});

export const cancelLogin = () => ({
  type: CANCEL_LOGIN,
});