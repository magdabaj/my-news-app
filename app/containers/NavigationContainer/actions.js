/*
 *
 * NavigationContainer actions
 *
 */

import { GET_TAGS, GET_ARTICLES_BY_TAG, GET_TAGS_FAILED, GET_TAGS_SUCCESS } from "./constants";

export const getTags = () => ({
  type: GET_TAGS
});

export const getTagsSuccess = tags => ({
  type: GET_TAGS_SUCCESS,
  tags
});

export const getTagsFailed = error => ({
  type: GET_TAGS_FAILED,
  error
});

export const getArticlesByTag = tagId => ({
  type: GET_ARTICLES_BY_TAG,
  tagId,
});
