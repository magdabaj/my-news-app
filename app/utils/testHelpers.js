import { Provider } from 'react-redux'
import React from 'react'
import { Router, BrowserRouter, browserHistory } from 'react-router-dom'
import configureStore from '../configureStore'

export const withStoreIntlAndRouter = component => {
    const store = configureStore({}, browserHistory)
    return (
        <Provider store={store}>
                    <BrowserRouter>{component}</BrowserRouter>
        </Provider>
    )
}