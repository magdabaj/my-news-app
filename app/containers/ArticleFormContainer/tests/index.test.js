/**
 *
 * Tests for ArticleFormContainer
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from "react";
import { render } from "react-testing-library";
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { ArticleFormContainer } from "../index";
import {addArticle, addArticleCanceled} from "../actions";
import {withStoreIntlAndRouter} from "../../../utils/testHelpers";

function renderArticleFormContainer(args) {
  const defaultprops = {
    addArticle: () => {},
    addArticleCanceled: () => {},
    loggedUser: {},
    selectedTag: {},
  };

  const props = {...defaultprops, ...args};

  return render(withStoreIntlAndRouter(<ArticleFormContainer {...props} />))
}

describe("<ArticleFormContainer />", () => {
  it("Expect to not log errors in console", () => {
    const spy = jest.spyOn(global.console, "error");
    renderArticleFormContainer();
    expect(spy).not.toHaveBeenCalled();
  });

  it("Expect to have additional unit tests specified", () => {
    expect(true).toEqual(true);
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it("Should render and match the snapshot", () => {
    const {
      container: { firstChild }
    } = renderArticleFormContainer();
    expect(firstChild).toMatchSnapshot();
  });
});
