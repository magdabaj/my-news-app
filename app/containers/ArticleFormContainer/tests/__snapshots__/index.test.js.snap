// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleFormContainer /> Should render and match the snapshot 1`] = `
<div
  class="Heading-ez7kt7-0 krJHDj"
>
  You must login in order to add article
</div>
`;
